app/controllers/activities_controller.rb

```diff
@@ -0,0 +1,40 @@
+class ActivitiesController < ApplicationController
+  def index
+    @activities = appointment.activities.since(since)
+
+    if @activities.present?
+      render partial: @activities
+    else
+      head :not_modified
+    end
+  end
+
+  def create
+    @activity = MessageActivity.create(message_params)
+
+    respond_to do |format|
+      format.js { render @activity }
+    end
+  end
+
+  private
+
+  def message_params
+    params
+      .require(:message_activity)
+      .permit(:message)
+      .merge(
+        user: current_user,
+        appointment: appointment
+      )
+  end
+
+  def since
+    timestamp = params[:timestamp].to_i / 1000
+    Time.zone.at(timestamp)
+  end
+
+  def appointment
+    current_user.appointments.find(params[:appointment_id])
```

Comment:
The appointments will be scoped to `current_user` so this will break things in multiple ways here. I don't think we need to limit who can comment on what outside of any other enforced permissions so I reckon we could just drop the scoping here.